src/Symfony/Component/Validator/Constraints/JsonValidator.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @author Imad ZAIRIG <imadzairig@gmail.com>
+ */
+class JsonValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Json) {
+            throw new UnexpectedTypeException($constraint, Json::class);
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!is_scalar($value) && !(\is_object($value) && method_exists($value, '__toString'))) {
```

Comment:
I don't feel like we need to support objects here. Just supporting strings looks best to me.

Reply:
is there any reasoning behind that? i'd prefer consistent behavior across all string constraints.

Reply:
Do we support objects in so many places? I thought that was only the case in a few.

Reply:
https://github.com/search?q=__toString+repo%3Asymfony%2Fsymfony+path%3Asrc%2FSymfony%2FComponent%2FValidator%2FConstraints&type=Code :)

Reply:
https://api.symfony.com/4.0/Symfony/Component/HttpFoundation/Response.html#method___toString (inherited by `JsonResponse`)

 i think the general consensus is to support value objects out-of-the box, i dont we should vary the behavior per constraint.